cmd/lotus-provider/guidedSetup/guidedSetup.go

```diff
@@ -0,0 +1,372 @@
+			fmt.Println(sty.Render(msg))
+		}
+}
+
+var migrationSteps = []migrationStep{
+	readMinerConfig, // Tells them to be on the miner machine
+	yugabyteConnect, // Miner is updated
+	verifySectors,   // Verify the sectors are in the database
+	configToDB,      // TODO work on base configuration migration.
+	doc,
+	oneLastThing,
+}
+
+type MigrationData struct {
+	T               func(key message.Reference, a ...interface{}) string
+	say             func(style lipgloss.Style, key message.Reference, a ...interface{})
+	selectTemplates *promptui.SelectTemplates
+	MinerConfigPath string
+	MinerConfig     *config.StorageMiner
+	*harmonydb.DB
+}
+
+func configToDB(d *MigrationData) {
+	d.say(section, "Migrating config.toml to database.")
+	d.say(plain, `A Lotus-Provider cluster shares a database. They share the work of proving multiple Miner ID's sectors.\n`)
+
+	type rawConfig struct {
+		Raw   []byte `db:"config"`
+		Title string
+	}
+	var configBytes []rawConfig
+	err := d.Select(context.Background(), &configBytes, `SELECT config, title FROM harmony_config `)
+	if err != nil {
+		d.say(notice, "Error reading from database: %s. Aborting Migration.\n", err.Error())
+		os.Exit(1)
+	}
+	//configs := lo.SliceToMap(configBytes, func(t rawConfig) (string, bool) {
+	// TODO
+	//})
+	// NEEDS multiaddress PR committed to interpret configs correctly.
+
+	// TODO	d.say(plain, "This lotus-miner services the Miner ID: %s\n", "TODO MinerID")
+
+	// This will be added to the new/existing base layer along with its specific wallet rules.
+	// (if existing): Here's the diff of this miner's config.toml and the db's base layer.
+	//      To edit, use the interactive editor by doing ....  after setup completes.
+	// (if new): Writing the base layer.
+
+	// commit the new base layer, also commit a miner-id-named layer.
+	d.say(plain, "TODO FINISH THIS FUNCTION\n")
+}
+
+func oneLastThing(d *MigrationData) {
+	d.say(section, "To bring you the best SP tooling...")
+	d.say(plain, "Share with the Curio team your interest in lotus-provider for this Miner ID.\n")
+	d.say(plain, "Hit return to tell http://CurioStorage.org that you've migrated to lotus-provider.\n")
+	_, err := (&promptui.Prompt{Label: "Press return to continue"}).Run()
+	if err != nil {
+		d.say(notice, "Aborting remaining steps.\n", err.Error())
+	}
+	// TODO http call to home with a simple message of
+	// this version, net (mainnet/testnet), and "lotus-provider" signed by the miner's key.
+}
+
+func doc(d *MigrationData) {
+	d.say(plain, "The configuration layers have been created for you: base, post, gui, seal.")
+	d.say(plain, "Documentation: \n")
+	d.say(plain, "Put common configuration in 'base' and include it everywhere.\n")
+	d.say(plain, "Instances without tasks will still serve their sectors for other providers.\n")
+	d.say(plain, "As there are no local config.toml files, put per-machine changes in additional layers.\n")
+	d.say(plain, "Edit a layer with the command: ")
```

Comment:
This doc seems to lack some context. It is easy for me to understand as I have the context of how provider works but that may not be the case for new users.